h2o-genmodel/src/main/java/hex/genmodel/algos/coxph/CoxPHMojoModel.java

```diff
@@ -0,0 +1,75 @@
+package hex.genmodel.algos.coxph;
+
+import hex.genmodel.MojoModel;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+public class CoxPHMojoModel extends MojoModel  {
+  double[] _coef;
+  String[] _coef_names;
+  int _numStart;
+  Map<List<Integer>, Integer> _strata;
+  int _strata_len;
+  double[][] _x_mean_cat;
+  double[][] _x_mean_num;
+  private int[] _coef_indexes;
+
+  CoxPHMojoModel(String[] columns, String[][] domains, String responseColumn) {
+    super(columns, domains, responseColumn);
+
+  }
+
+  void prepareCoefIndexes() {
+    final List<String> namesInList = Arrays.asList(_modelDescriptor.columnNames());
+    _coef_indexes = new int[_coef_names.length];
+    int i = 0;
+    for (String coefName : _coef_names) {
+      Integer index = namesInList.indexOf(coefName);
+      _coef_indexes[i++] = index;
+    }
+  }
+
+  @Override
+  public double[] score0(double[] row, double[] predictions) {
+    if (null == _coef_indexes) {
```

Comment:
ideally Mojo should be "state-less", any reason this cannot be done in constructor? or some post-init function called by the reader?

Reply:
Does it https://github.com/h2oai/h2o-3/commit/c87713ffaf45e3862e4cc24dd6aa2d16c2e3d743 solve for you?

Reply:
I was thinking more of a package private method used by coxph only, adding public method to GenModel kindof kills the purpose, since you could expect this would get called only once

Reply:
I was thinking about this before, but `readAll` is private and it calls `readModelData` too early (before `      _model._modelAttributes` is available. And `readFrom` is `static`